File: src/Scripting/Lua/bridge.lua
local function device(name)
  return {
    turn_on = function() return name .. ':on' end,
    mute = function() return name .. ':muted' end
  }
end

local function remote(action)
  return function(d) return action(d) end
end

local basic = remote(function(d) return d.turn_on() end)
local mute = remote(function(d) return d.mute() end)
local tv = device('TV')
local radio = device('Radio')

print('basic-tv=' .. basic(tv))
print('basic-radio=' .. basic(radio))
print('mute-tv=' .. mute(tv))
print('mute-radio=' .. mute(radio))

File: src/Scripting/Perl/bridge.pl
use strict;
use warnings;

sub device {
    my ($name) = @_;
    return {
        turn_on => sub { return "$name:on" },
        mute    => sub { return "$name:muted" },
    };
}

sub basic_remote {
    my ($device) = @_;
    return $device->{turn_on}->();
}

sub mute_remote {
    my ($device) = @_;
    return $device->{mute}->();
}

my $tv = device('TV');
my $radio = device('Radio');
print 'basic-tv=' . basic_remote($tv) . "\n";
print 'basic-radio=' . basic_remote($radio) . "\n";
print 'mute-tv=' . mute_remote($tv) . "\n";
print 'mute-radio=' . mute_remote($radio) . "\n";
